### PR TITLE
Suppress the struct==null warning except in "strict" mode

### LINKED ISSUE #663: Roll back warning for struct == null
Roslyn added a new warning compared to the native compiler: if you attempt to compare a struct value to null
```cs
  if (structValue == null) ...
```
Roslyn now gives you a warning (the value in this case is always false). It still typechecks because the comparison operator selected is `operator==(Struct? left, Struct? right)`.

Unfortunately this is a breaking change for someone who instructs the compiler to treat warnings as errors. So the warning should be removed.

In the longer run we want a mechanism to add warnings in which users would be required to opt-in before receiving any new warnings. When that is done this warning can be restored as one you'd need to opt in to. But that is a separate issue.

### PR DESCRIPTION
**Pull request by gafter**
Fixes #663

### CODE REVIEW

**Comment by gafter** [comment]
@jaredpar @VSadov @AlekseyTs @agocke @VladimirReshetnikov Could a couple of you review this please?

**Comment by jaredpar** [comment]
LGTM

File: src/Compilers/Test/Utilities/CSharp/TestOptions.cs [hunk 1]
```diff
@@ -32,5 +34,10 @@ public static class TestOptions
 
         public static readonly CSharpCompilationOptions UnsafeDebugDll = DebugDll.WithAllowUnsafe(true);
         public static readonly CSharpCompilationOptions UnsafeDebugExe = DebugExe.WithAllowUnsafe(true);
+
+        public static CSharpCompilationOptions WithStrictMode(this CSharpCompilationOptions options)
+        {
+            return options.WithFeatures(ImmutableArray.Create("strict"));
```
Comment (AlekseyTs) [inline src/Compilers/Test/Utilities/CSharp/TestOptions.cs]: Is this additive? I would expect WithStrictMode helper to preserve whatever features already set.

File: src/Test/Utilities/DiagnosticDescription.cs [hunk 2]
```diff
@@ -192,6 +192,8 @@ public DiagnosticDescription WhereSyntax(Func<SyntaxNode, bool> syntaxPredicate)
             return new DiagnosticDescription(_code, _isWarningAsError, _squiggledText, _arguments, _startPosition, syntaxPredicate, _argumentOrderDoesNotMatter, _errorCodeType);
         }
 
+        public object Code => _code;
```
Comment (AlekseyTs) [inline src/Test/Utilities/DiagnosticDescription.cs]: This looks like a public API change. Did it go through the proper approval process? If it was approved, please exclude it from this pull request and submit as a separate change.

Comment (gafter) [inline src/Test/Utilities/DiagnosticDescription.cs]: This class is a test helper and not in the Roslyn public API.

File: src/Compilers/Test/Utilities/CSharp/TestOptions.cs (hunk 1 above)
Comment (gafter) [inline src/Compilers/Test/Utilities/CSharp/TestOptions.cs]: As it turns out, a test helper that preserves other feature flags wasn't needed, as there are no other features that we test at the same time.

Comment (AlekseyTs) [inline src/Compilers/Test/Utilities/CSharp/TestOptions.cs]: Then the name of the method is confusing and should be changed to make it clear that it overwrites Features.

File: src/Test/Utilities/DiagnosticDescription.cs (hunk 2 above)
Comment (AlekseyTs) [inline src/Test/Utilities/DiagnosticDescription.cs]: I see. Thanks.

**Comment by gafter** [comment]
@AlekseyTs I believe your review comments have been addressed.

**Comment by agocke** [comment]
LGTM

**Comment by AlekseyTs** [comment]
LGTM

**Pull request merged**